tests/queries/0_stateless/02346_gin_index_definition.sql

```diff
@@ -0,0 +1,151 @@
+SET allow_experimental_full_text_index = 1;
```

Comment:
I think we will also need to adjust all other Gin index test files since we are no longer backward compatible.

Please consider also unifying existing tests 02346_gin_index_creation_negative with the new test. They more or less check the same functionality.